src/redux-enhancers/undoRedoReducerEnhancer.ts

```diff
@@ -4,9 +4,30 @@ import { Action, Store, StoreEnhancer, StoreEnhancerStoreCreator } from 'redux'
 import { NAVIGATION_ACTIONS, UNDOABLE_ACTIONS } from '../constants'
 import { State } from '../util/initialState'
 import { Index, Patch } from '../types'
+import { updateThoughts } from '../reducers'
 
 const stateSectionsToOmit = ['alert', 'pushQueue', 'user']
 
+/**
+ * Manually extract thought and context index updates along with pushQueue.
+ */
+const extractUpdates = (patch: Patch, newState: State) => {
```

Comment:
Think of any function that takes a `State` and returns a new `State` as a reducer. `state: State` should be the first argument by convention.